internal/ethapi/tracer.go

```diff
@@ -240,21 +232,20 @@ func NewJavascriptTracer(code string, timeout time.Duration) (*JavascriptTracer,
 		db:         db,
 		dbvalue:    db.toValue(vm),
 		err:        nil,
-		timeout:    timeout,
 	}, nil
 }
 
+// Stop terminates execution of any JavaScript
+func (jst *JavascriptTracer) Stop(err error) {
+	jst.err = err
+	jst.vm.Interrupt <- func() {
+		panic(err)
+	}
+}
```

Comment:
I don't see why. Two assignments in a race will result in one or the other winning; in either case we correctly report one of the error(s) that occurred to the user.